promql/engine.go

```diff
@@ -218,22 +218,27 @@ func contextDone(ctx context.Context, env string) error {
 // Engine handles the lifetime of queries from beginning to end.
 // It is connected to a querier.
 type Engine struct {
-	// The querier on which the engine operates.
-	querier local.Querier
+	// A Querier constructor against an underlying storage.
+	queryable Queryable
```

Comment:
I think they are equivalent for all practical purposes currently – just passing in the storage being simpler to use. The function approach requires changing dozens of occurrences across regular and testing code, which I wasn't keen to introduce.